### PR TITLE
[rest.li] Add util class to convert generic List/Map to DataList/DataMap or vice versa

### PR DESCRIPTION
**Pull request by vshwnth2**
Currently, there are some internal implementations of this, but logic is pretty standard conversion between java and restli objects.

### CODE REVIEW

File: data/src/main/java/com/linkedin/util/DataComplexUtil.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Utils to convert generic List/Map to DataList/DataMap or vice versa
+ */
+public class DataComplexUtil {
```
Comment (saponniah) [inline data/src/main/java/com/linkedin/util/DataComplexUtil.java:18]: Can you move the relevant unit test from here?

https://git.corp.linkedin.com:1367/a/plugins/gitiles/multiproducts/graphql-infra/+/master/util/src/test/java/com/linkedin/graphql/DataComplexUtilTest.java

File: data/src/main/java/com/linkedin/util/DataComplexUtil.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+  private DataComplexUtil() { }
+
+  /**
+   * Converts the input {@link Map} to a {@link DataMap} recursively.
+   * Optional to stringfy the value of the leaf nodes in the datacomplex
+   */
+  public static DataMap convertMap(Map<String, ?> input, boolean stringify) {
```
Comment (aman1309) [inline data/src/main/java/com/linkedin/util/DataComplexUtil.java:26]: It's annoying but can follow the Restli repo indentation ie '{' in a new line

**Review by saponniah — APPROVED** [review]

**Comment by junchuanwang** [comment]
please add changelog and upgrade pegasus version?

**Review by junchuanwang — CHANGES_REQUESTED** [review]
please add changelog and upgrade pegasus version?

File: data/src/main/java/com/linkedin/util/DataComplexUtil.java (hunk 2 above)
Comment (junchuanwang) [reply data/src/main/java/com/linkedin/util/DataComplexUtil.java]: yeah it is call AllMan's style. https://en.wikipedia.org/wiki/Indentation_style

Java code doesn't usually do that though

**Review by junchuanwang — APPROVED** [review]

**Pull request merged**